instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -446,12 +449,67 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
                 .navigationRoutes
 
             assertEquals(224.224, requestedRoutes[0].getSumOfDurationAnnotationsFromLeg(0), 0.0001)
-            assertEquals(169.582, refreshedRoutes[0].getSumOfDurationAnnotationsFromLeg(0), 0.0001)
+            assertEquals(172.175, refreshedRoutes[0].getSumOfDurationAnnotationsFromLeg(0), 0.0001)
 
             assertEquals(227.918, requestedRoutes[1].getSumOfDurationAnnotationsFromLeg(0), 0.0001)
-            assertEquals(234.024, refreshedRoutes[1].getSumOfDurationAnnotationsFromLeg(0), 0.0001)
+            assertEquals(235.641, refreshedRoutes[1].getSumOfDurationAnnotationsFromLeg(0), 0.0001)
         }
 
+    @Test
+    fun expect_route_refresh_to_update_annotations_for_new_alternative() = sdkTest {
+        setupMockRequestHandlers(
+            twoCoordinates,
+            R.raw.route_response_single_route_multileg,
+            R.raw.route_response_single_route_multileg_refreshed,
+            "route_response_single_route_multileg",
+            acceptedGeometryIndex = 70
+        )
+        val routeOptions = generateRouteOptions(twoCoordinates)
+        val requestedRoutes = mapboxNavigation.requestRoutes(routeOptions)
+            .getSuccessfulResultOrThrowException()
+            .routes
+        val alternativeRoute =
+            RoutesProvider.alternative_for_multileg(activity).toNavigationRoutes().first()
+
+        mapboxNavigation.setNavigationRoutes(requestedRoutes, initialLegIndex = 1)
+        mapboxNavigation.startTripSession()
+
+        // corresponds to currentRouteGeometryIndex = 70 for primary route and 11 for alternative route
+        stayOnPosition(38.581798, -121.476146)
+        mapboxNavigation.routeProgressUpdates()
+            .filter {
+                it.currentRouteGeometryIndex == 70
+            }
+            .first()
+
+        mockWebServerRule.requestHandlers.add(
+            FailByRequestMockRequestHandler(
+                MockDirectionsRefreshHandler(
+                    "route_response_alternative_for_multileg",
+                    readRawFileText(
+                        activity,
+                        R.raw.route_response_alternative_for_multileg_refreshed
+                    ),
+                    acceptedGeometryIndex = 11
+                )
+            )
+        )
```

Comment:
why do you add the header in the middle of the test? Is that because the last header you add takes over all route refresh requests?

Reply:
Not really. It just doesn't have any effect until we add an alternative. Moved it to the beginning of the test.